assets/js/translator.js

```diff
@@ -58,12 +58,10 @@ if(modeEnabled('translation')) {
             });
 
             $('#originalText').on('input propertychange', function () {
-                var disableDetect = this.value === '';
-                $('#detect, #srcLangSelect option[value="detect"]').prop('disabled', disableDetect);
-                $('#detect').toggleClass('disabledLang', disableDetect);
-
-                persistChoices('translator');
+                updateDetect($('#originalText').text() !== '');
```

Comment:
What would happen if this triggers with `false` here and then `true`? There's no `true` call corresponding to your `$('#detect, #srcLangSelect option[value="detect"]').prop('disabled', true);`.

Reply:
Interesting, none of the previous code did this properly.